conmon-rs/server/src/oom_watcher.rs

```diff
@@ -0,0 +1,270 @@
+use anyhow::{anyhow, bail, Context, Result};
+use lazy_static::lazy_static;
+use log::{debug, error};
+use nix::sys::statfs::{statfs, FsType};
+use notify::{Error, Event, RecommendedWatcher, RecursiveMode, Watcher};
+use regex::Regex;
+use std::os::unix::prelude::AsRawFd;
+use std::path::{Path, PathBuf};
+use tokio::fs::File;
+use tokio::io::{AsyncBufReadExt, AsyncReadExt, AsyncWriteExt, BufReader};
+use tokio::sync::broadcast::Receiver as BroadcastReceiver;
+use tokio::sync::mpsc::{channel, Receiver};
+use tokio_eventfd::EventFd;
+
+use crate::child_reaper::ExitChannelData;
+
+#[cfg(any(all(target_os = "linux", target_env = "musl")))]
+pub const CGROUP2_SUPER_MAGIC: FsType = FsType(libc::CGROUP2_SUPER_MAGIC as u64);
+#[cfg(any(all(target_os = "linux", not(target_env = "musl"))))]
+pub const CGROUP2_SUPER_MAGIC: FsType = FsType(libc::CGROUP2_SUPER_MAGIC as i64);
+
+static CGROUP_ROOT: &str = "/sys/fs/cgroup";
+
+pub struct OOMWatcher {}
+
+impl OOMWatcher {
+    /// this function blocks until the exit_rx channel has closed
+    pub async fn wait_for_oom(
+        exit_rx: &mut BroadcastReceiver<ExitChannelData>,
+        pid: u32,
+        exit_paths: &[PathBuf],
+    ) -> Result<()> {
+        match statfs(CGROUP_ROOT) {
```

Comment:
nit, as a follow up we should detect this at startup and and save an impl internally, rather than checking and matching each time

Reply:
fixed this